lib/command.js

```diff
@@ -1126,6 +1116,30 @@ Expecting one of '${allowedValues.join("', '")}'`);
   }
 
   _prepareForParse() {
+    // Before parsing and before saving the state, do the special default of lone negated option to true.
+    // Some paranoia about not overwriting a value since doing this lazily instead of when option added.
+    if (this._savedState === null) {
+      this.options
+        .filter(
+          (option) =>
+            option.negate &&
+            option.defaultValue === undefined &&
+            this.getOptionValue(option.attributeName()) === undefined,
+        )
+        .forEach((option) => {
+          // lone --no-foo is special and defaults foo to true, unless a --foo option is already defined
+          const positiveLongFlag = option.long.replace(/^--no-/, '--');
+          if (!this._findOption(positiveLongFlag)) {
+            this.setOptionValueWithSource(
+              option.attributeName(),
+              true,
+              'default',
+            );
+          }
+        });
+    }
+
+    // Save the state the first time, then restore the state before each subsequent parse.
     if (this._savedState === null) {
```

Comment:
Since the conditon is the same, how about not separating them so it is easier to understand?

```suggestion
    if (this._savedState === null) {
      // Before parsing and before saving the state, do the special default of lone negated option to true.
      // Some paranoia about not overwriting a value since doing this lazily instead of when option added.
      this.options
        .filter(
          (option) =>
            option.negate &&
            option.defaultValue === undefined &&
            this.getOptionValue(option.attributeName()) === undefined,
        )
        .forEach((option) => {
          // lone --no-foo is special and defaults foo to true, unless a --foo option is already defined
          const positiveLongFlag = option.long.replace(/^--no-/, '--');
          if (!this._findOption(positiveLongFlag)) {
            this.setOptionValueWithSource(
              option.attributeName(),
              true,
              'default',
            );
          }
        });
      // Save the state the first time, then restore the state before each subsequent parse.
```

Reply:
I thought it might be useful having the simple save/restore together, but didn't really help. Refactored as suggested!